crypto/eddsa.go

```diff
@@ -0,0 +1,78 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+// Ed25519GenerateKey returns a random PEM encoded Ed25519 Private Key.
+func Ed25519GenerateKey() ([]byte, error) {
+	_, secret, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("Ed25519GenerateKey: failed to generate EDDSA private key: %w", err)
+	}
+	return pemEncodeEdPrivateKey(secret)
+}
+
+// Ed25519GenerateKeyFromSeed returns a PEM encoded Ed25519 Private Key from
+// `seed`. Panics if len(seed) is not SeedSize.
+func Ed25519GenerateKeyFromSeed(seed []byte) ([]byte, error) {
+	return pemEncodeEdPrivateKey(ed25519.NewKeyFromSeed(seed)) // Panics
```

Comment:
This should not panic. Can you add a guard around `seed` to check its length first, and return an error instead?

Reply:
Added a guard.